modules/300-prometheus/templates/prometheus/prometheus.yaml

```diff
@@ -55,6 +55,9 @@ spec:
   enableRemoteWriteReceiver: true
   enableFeatures:
   - memory-snapshot-on-shutdown # https://ganeshvernekar.com/blog/prometheus-tsdb-snapshot-on-shutdown/
+{{- if .Values.prometheus.externalLabels }}
+  - expand-external-labels # https://prometheus.io/docs/prometheus/latest/feature_flags/#expand-environment-variables-in-external-labels
+{{- end }}
```

Comment:
1. This should not be required because the Prometheus pod has a config reloader sidecar that does the job. Probably, the flag is not required.
2. If a flag is still required, it should always be enabled. Otherwise, if an external label is added to the module config, Prometheus will be restarted.

Reply:
1. The flag is required, without it Prometheus doesn't do variables expansion.
2. Enabling the flag on condition was nothing more than adhering to the principle of least privilege.

Reply:
1. Yes, without the flag Prometheus cannot do this, but `config-reloader` sidecar does the trick. 
See [this](https://github.com/prometheus-operator/prometheus-operator/blob/main/cmd/prometheus-config-reloader/main.go) and [this](https://github.com/thanos-io/thanos/blob/main/pkg/reloader/reloader.go) for more details.
2. Restarting the Prometheus is more problematic than enabling it in the pod by default.

Reply:
The flag has been removed and now config-reloader does env substitutions.